src/handledRejectionPromise.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Use this class if you have a promise that might be rejected and that you're
+ * not "always" awaiting (or using then).
+ * E.g. Consider you're creating a promise that might reject, but only awaiting
+ * it later (when it's actually being used). If that promise is rejected
+ * between promise creation and promise awaiting, then node.js treats this as
+ * "uncaught exception" and immediately exists with exit code 1. To avoid this,
+ * just wrap that promise into an HandledRejectionPromise.
+ *
+ * It is not allowed to use this class to silently ignore exceptions.
+ */
+export class HandledRejectionPromise<T> {
+  private storedPromise: Promise<void>;
+  private value:
+    | undefined
+    | { type: "resolved"; value: T }
+    | { type: "rejected"; reason: unknown };
+
+  constructor(promise: Promise<T>) {
+    this.storedPromise = promise.then(
+      (value) => {
+        this.value = { type: "resolved", value };
+        return;
+      },
+      (reason: unknown) => {
+        this.value = { type: "rejected", reason };
+      }
+    );
```

Comment:
I don't see a reason for not using async await syntax here, same below for the promise getter

Reply:
Yeah, that's what I had in mind, but I understand that one can find it "iffy" :D

will merge it tomorrow or on monday or so

Reply:
:D

Shall I replace the `get promise()` with `async promise()`?

Reply:
no, it's fine :)